### ISSUE TITLE
Building from source yields degraded XPI

### ISSUE THREAD

**Issue by madduck** [passff/passff#355]
I was debugging #354 and installed a `passff.xpi` from file. Since then, even after reverting to the previous XPI, or installing the current one from AMO, and restarting Firefox etc., I cannot fill in login forms from the `ctrl-y` menu anymore. The menu pops up, but if I choose any of the "Fill" options, simply nothing happens.

However, filling in still works fine using the (P) buttons, as well as the context menu.

The status line says "[21:24:03] show -> (0) no error message".

And I cannot see anything on the console, even after enabling logging.

**Comment by tuxor1337** [comment]
Please create a new Firefox profile, try to reproduce the issue from there and give step by step instructions on how to reproduce.

**Comment by madduck** [comment]
I am sorry that I forgot to mention that I had already tried this, and I cannot reproduce the problem, of course. I would much rather figure out where the problem lies now, than to rebuild my entire profile from scratch, because this might happen again, and rebuilding is just not a solution.

**Comment by madduck** [comment]
On a related note, fill-submitting using the paper airplane in the (P) menu no longer works, but yields "invalid login credentials" on all sites I've tried. Filling with the pen, and hitting submit myself, however, still works.

So something got really messed up, but it's not the PassFF code. What could it be in the rest of Firefox that is breaking this so badly?

**Comment by tuxor1337** [comment]
I'm sorry to hear that you're having trouble using PassFF with your current Firefox profile. Unfortunately, the behaviour you experience is so strange that I don't immediately have an idea where to look.

Have you tried loading the plugin from the source directory using `about:debugging`?

**Comment by madduck** [comment]
Dear @tuxor1337, yeah, this is weird, but thank you so much for being patient with me. Probably, this will all be due to my own stupidity, but I raked my brain, and I cannot seem to figure out what's going on. Plus, I really only just installed a different version of PassFF, and reverted, and now this problem exists.

Thank you for teaching me how to use the debugger. This really helps. So I disabled all other extensions, closed all tabs, and can still reproduce the problem: If I try to fill the form from the menu, nothing happens. But on the console, the following error is logged:

```
[Exception... "Component returned failure code: 0xc1f30001 (NS_ERROR_NOT_INITIALIZED) [nsIMessageSender.sendAsyncMessage]"  nsresult: "0xc1f30001 (NS_ERROR_NOT_INITIALIZED)"                   (unknown)
location: "JS frame :: resource://gre/modules/ExtensionUtils.jsm :: sendAsyncMessage :: line 533"  data: no]

	sendAsyncMessage                                                                                                                              resource://gre/modules/ExtensionUtils.jsm:533:51
	_handleMessage/deferred.promise<                                                                                                               resource://gre/modules/MessageChannel.jsm:984:9
```

If I use the `(P)` menu to fill&submit, the username is filled, as is a password, but I am told that the credentials are wrong. I will return to this in a separate issue, once we resolved this (if the issue then still persists, I feel like the two are related).

So now on to debugging, but I am not having much luck here either. For a start, I set a breakpoint at [modules/menu.js:156](https://github.com/passff/passff/blob/master/src/modules/menu.js#L156), the debugger seems to fall over once I hit `<F10>`, with the following error:

```
error occurred while processing 'resume: TypeError: can't access dead object                                                                                 main.js:1624
Stack: createNonSourceMappedActor@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/utils/TabSources.js:312:1
getFrameLocation@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/utils/TabSources.js:588:7
_handleResumeLimit@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/thread.js:571:31
onResume@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/thread.js:673:28
onPacket@resource://devtools/shared/base-loader.js -> resource://devtools/server/main.js:1769:15
receiveMessage@resource://devtools/shared/base-loader.js -> resource://devtools/shared/transport/transport.js:760:7
enter@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/utils/event-loop.js:118:5
_pushThreadPause@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/thread.js:182:5
_pauseAndRespond@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/thread.js:397:7
hit@resource://devtools/shared/base-loader.js -> resource://devtools/server/actors/breakpoint.js:188:12
keyPressManagement@moz-extension://2b3f7659-6f47-4cec-b523-bd76547d9877/modules/menu.js:156:1
onSearchKeypress@moz-extension://2b3f7659-6f47-4cec-b523-bd76547d9877/modules/menu.js:119:5
Line: 312, column: 1
```

This seems to be a problem with my debugger, as it doesn't seem to matter where I put the breakpoint, and the problem seems to be within devtools. FWIW, the scope watching pane in the debugger shows all the PassFF variables defined in the scope, and there is certainly no problem with the line `if (event.keyCode == 13)`, i.e. `event` is well-defined, and `event.keyCode` is even 13…

I'll keep looking, but wanted to keep a log of my progress, so updating this issue here.

**Comment by madduck** [comment]
I succeeded in creating a clean profile, where I only modified the configuration to allow local XPIs to be installed, installed my custom XPI file, and am able to reproduce the problem. Here is a [ZIP file](https://scratch.madduck.net/__tmp__debug-passff.zip) with the profile, as well as the checkout of the PassFF source.

**Comment by madduck** [comment]
I cannot reproduce the problem in a clean profile, with the XPI from AMO. So the problem is definitely with the lcoally built XPI file. However, please note that this used to work just fine for months, until I made a modification as suggested in #354, rebuilt, and eventually rolled back the change, removed the `bin` and `build` directories from the source tree, and ran `make -C src` again. I then installed the XPI from `bin/testing/passff.xpi`, and since then I've been having that problem. Actually the problem may have been present already in the test build addressing the problem of #354, but I do not remember.

Maybe I am building the XPI wrongly?

Here is the output from what happens when I build:

```
fishbowl:…sswordstore/passff|master|% rm -rf build/ bin/
fishbowl:…sswordstore/passff|master|% make -C src
make: Entering directory '/home/madduck/code/passwordstore/passff/src'
mkdir -p ../build/testing
cp -r ./content ../build/testing/content
cp -r ./_locales ../build/testing/_locales
cp -r ./modules ../build/testing/modules
cp -r ./skin ../build/testing/skin
cp -r ./icon.png ../build/testing/icon.png
cp ./manifest.json ../build/testing/manifest.json
sed -i "s/_VERSIONHOLDER_/testing/g" ../build/testing/manifest.json
mkdir -p ../bin/testing
rm -f ../bin/testing/passff.xpi
cd ../build/testing; zip __.zip -q -r ./* -x \*/.d
mv ../build/testing/__.zip ../bin/testing/passff.xpi
make: Leaving directory '/home/madduck/code/passwordstore/passff/src'
```

**Comment by tuxor1337** [comment]
This might be a bug introduced by commit fd7f8097353f35e1b259ded56e3a248611d24590 which is not in AMO so far. I addressed this in 233024d. Please try to reproduce your issue using code that includes the new bug fix.

**Comment by madduck** [comment]
Yes, that seems to fix the problem. Thanks for chasing this down so quickly!

Closing, as this only affects manual builds, and `HEAD` works now.
